m3/src/main/java/com/uber/m3/tally/m3/M3Reporter.java

```diff
@@ -217,7 +259,7 @@ public void close() {
 
         // Important to use `shutdownNow` instead of `shutdown` to interrupt processor
         // thread(s) or else they will block forever
-        executor.shutdownNow();
+        executorService.shutdownNow();
```

Comment:
add shutdown for scheduledExecutorService?

Reply:
Good catch